File: Add-Strings/AS.rs
use std::collections::HashMap;

fn exponent(exp: i32) -> i32 {
    let mut result = 1;
    for _ in 0..exp {
        result *= 10;
    }
    result
}

fn add_strings(nums1: &str, nums2: &str, dict: &HashMap<char, i32>) -> i32 {
    let length1 = nums1.len() as i32;
    let length2 = nums2.len() as i32;
    let mut sum = 0;
    let mut count = 1;

    for c in nums1.chars() {
        //:= MARK: well, (expression) wont take ownship
        sum += *dict.get(&c).unwrap() * exponent((length1 - count));
        count += 1;
        //println!("{}", &sum);
    }

    count = 1;
    for c in nums2.chars() {
        sum += *dict.get(&c).unwrap() * exponent((length2 - count));
        count += 1;
        //println!("{}", &sum);
    }

    sum
}

fn main() {
    let mut dict = HashMap::new();
    let mut temp_num = 0;

    for c in "0123456789".chars() {
        dict.insert(c, temp_num);
        temp_num += 1;
    }

    println!("{}", add_strings(&"102", &"20", &dict));
}

File: Add-Strings/AS.jl
function addStrings(s1::String, s2::String) :: Int64
    chars = ['0','1','2','3','4','5','6','7','8','9']
    dd = Dict([(chars[n+1],n) for n in 0:9])

    result = 0
    len1 = length(s1)
    for ind in 1:len1
        result += dd[s1[ind]] * (10 ^ len1)/(10 ^ ind)
    end

    len2 = length(s2)
    for ind in 1:len2
        result += dd[s2[ind]] * (10 ^ len2)/(10 ^ ind)
    end
    
    result
end
